packages/zoe/src/contracts/loan/updateDebt.js

```diff
@@ -46,32 +50,53 @@ export const makeDebtCalculator = debtCalculatorConfig => {
 
   const config = { ...configMinusGetDebt, getDebt };
 
-  const updateDebt = _state => {
-    const interest = loanMath.make(calcInterestFn(debt.value, interestRate));
-    debt = loanMath.add(debt, interest);
-    debtNotifierUpdater.updateState(debt);
+  const updateDebt = timestamp => {
+    let prevUpdateTimestamp = lastCalculationTimestamp;
+    while (prevUpdateTimestamp + interestPeriod <= timestamp) {
+      prevUpdateTimestamp += interestPeriod;
+      const interest = loanMath.make(calcInterestFn(debt.value, interestRate));
+      debt = loanMath.add(debt, interest);
+      debtNotifierUpdater.updateState(debt);
```

Comment:
Should this be inside or outside the loop?  I would expect it to be outside to roll-up the updates into a single update.